Add first look guides (#74)
neo4j/docs-ops-manager: modules/ROOT/pages/first-look/linux-first-look.adoc

== Create persistence DBMS

=== Download and extract
Download latest Neo4j Linux/Mac executable from https://neo4j.com/download-center/#ops-manager[here] and extract the package. 
Extracted directory location will be referred to as NEO4J_HOME form now on.

=== Change password
Navigate to NEO4j_HOME and run:

[source, terminal, role=noheader]
----
./bin/neo4j-admin dbms set-initial-password lovelypassword
----

=== Run Neo4j 
[source, terminal, role=noheader]
----
./bin/neo4j start
----

See link:https://neo4j.com/docs/operations-manual/current/installation/linux/tarball/[Neo4j Linux installation documentation] for more details about installing and running Neo4j.

== Server installation
=== Unpack NOM server bundled with Neo4j
   
[source, terminal, role=noheader]
----
   tar -xzf NEO4J_HOME/products/neo4j-ops-manager-server-*-unix.tar.gz
----
    
Extracted directory will now be known as NOM_SERVER_HOME.

=== Generate self-signed certificates

Navigate to NOM_SERVER_HOME and run:

[source, terminal, role=noheader]
----
java -jar ./lib/server.jar ssc -n localhost \
        -o ./certificates \

>         -p secretpassword \

Reviewer: Can we make the secretpassword stuff stand out? Maybe in the text before? Or are we okay with them generating certs with that password? I know this is only for eval/demo use but maybe it would be wise to not have a hard-coded secret in case people actually wrongly use it in production

Author: Good point - I think we can make it bold and wrap it in angled brackets. AND make a note that it should be changed. Because you are right - someone will use this in production...

= Linux first look guide

== Prerequisites
 * Java 17
 * A Neo4j cluster meeting *xref:addition/instance-requirements.adoc[these]* requirements

NOTE: The query log feature is only available for DBMSs with instances that are running Neo4j 5.1.0 or higher.

== Create persistence DBMS

=== Download and extract
Download latest Neo4j Linux/Mac executable from https://neo4j.com/download-center/#ops-manager[here] and extract the package. 

> Extracted directory location will be referred to as NEO4J_HOME form now on.

Reviewer 2: ```suggestion
Extracted directory location is referred to as NEO4J_HOME form now on.
```

=== Change password
Navigate to NEO4j_HOME and run:

[source, terminal]
----
./bin/neo4j-admin dbms set-initial-password lovelypassword
----

=== Run Neo4j 
[source, terminal, role=noheader]
----
./bin/neo4j start
----

See link:https://neo4j.com/docs/operations-manual/current/installation/linux/tarball/[Neo4j Linux installation documentation] for more details about installing and running Neo4j.

== Server installation
=== Unpack NOM server from Neo4j package
   
[source, terminal]
----
   tar -xzf NEO4J_HOME/products/neo4j-ops-manager-server-*-unix.tar.gz
----
    

> Extracted directory will now be known as NOM_SERVER_HOME.

Reviewer 2: ```suggestion
Extracted directory is now known as NOM_SERVER_HOME.
```

====

More information on Self Signed certificates *xref:installation/self-signed-certificate.adoc[here]*.

=== Run NOM server as console application 

[source, terminal]
----
    java -jar ./lib/server.jar \
        --spring.neo4j.uri=neo4j: neo4j://localhost:7687\
        --spring.neo4j.authentication.username=neo4j \
        --spring.neo4j.authentication.password=lovelypassword \
        --server.port=8080 \
        --server.ssl.key-store-type=PKCS12 \
        --server.ssl.key-store=file:./certificates/localhost.pfx \
        --server.ssl.key-store-password=<PASSWORD> \
        --grpc.server.port=9090 \
        --grpc.server.security.key-store-type=PKCS12 \
        --grpc.server.security.key-store=file:./certificates/localhost.pfx\
        --grpc.server.security.key-store-password=<PASSWORD>\
        --jwt.secret=please-set-a-random-secret-string-here-for-jwt-signing \
----

Running as console application on Unix is documented *xref:installation/server.adoc#_unix[here]*

=== Open NOM UI
Wait for the server to start and then go to https://localhost:8080. 
Login as admin:passw0rd and accept license terms. 

== Set up agent
=== Register agent 

> In NOM UI - navigate to Agent settings (clicking on the sad robot will take you to the correct page) and add a new agent.

Reviewer 2: ```suggestion
In NOM UI - navigate to Agent settings (clicking on the sad robot takes you to the correct page) and add a new agent.
```

Copy environment variables that are provided. 

> Full documentation on registering an agent is *xref:addition/index.adoc#register[here]. 

Reviewer 3: ```suggestion
Full documentation on registering an agent is *xref:addition/index.adoc#register[here]*. 
```

=== Unpack agent
[source, terminal, role=noheader]
----
   tar -xzf NEO4J_HOME/products/neo4j-ops-manager-agent-*-linux-amd64.tar.gz
----
Extracted directory will now be known as NOM_AGENT_HOME.

=== Configure agent
Edit the export command below as follows: 

* Replace <OUTPUT_FROM_REGISTER_AGENT> with the four lines shown during the reigister agent step in the NOM UI. 

> * Replace the bolt URIs, usernames and passwords  with the correct ones for your cluster. 

Reviewer 2: ```suggestion
* Replace the bolt URIs, usernames and passwords with the correct ones for your cluster. 
```

CONFIG_INSTANCE_1_BOLT_USERNAME=neo4j \
CONFIG_INSTANCE_1_BOLT_PASSWORD=<SERVER_1_BOLT_PASSWORD> \
CONFIG_INSTANCE_1_QUERY_LOG_PORT=9500 \
CONFIG_INSTANCE_1_LOG_CONFIG_PATH=<SERVER1_HOME_DIR>/conf/server-logs.xml \
CONFIG_INSTANCE_2_NAME=server2 \
CONFIG_INSTANCE_2_BOLT_URI=<SERVER_2_BOLT_URI> \
CONFIG_INSTANCE_2_BOLT_USERNAME=neo4j \
CONFIG_INSTANCE_2_BOLT_PASSWORD=<SERVER_2_BOLT_PASSWORD> \
CONFIG_INSTANCE_2_QUERY_LOG_PORT=9600 \
CONFIG_INSTANCE_2_LOG_CONFIG_PATH=<SERVER2_HOME_DIR>/conf/server-logs.xml \
CONFIG_INSTANCE_3_NAME=server3 \
CONFIG_INSTANCE_3_BOLT_URI<SERVER_3_BOLT_URI> \
CONFIG_INSTANCE_3_BOLT_USERNAME=neo4j \
CONFIG_INSTANCE_3_BOLT_PASSWORD=<SERVER_3_BOLT_PASSWORD> \
CONFIG_INSTANCE_3_QUERY_LOG_PORT=9700 \
CONFIG_INSTANCE_3_LOG_CONFIG_PATH=<SERVER3_HOME_DIR>/conf/server-logs.xml
----
Run the edited export command. 

Full documentation on configuring an agent is *xref:addition/index.adoc#configure[here]*. 

=== Run agent as console application

[source, terminal]
----
./bin/agent console
----

== Explore NOM UI
Go to NOM UI and wait for DBMS to appear - this may take a few minutes. 
You should be able to see that the agent has connected in the agents listing. 

> Once DBMS is shown in home page, double click on the name (initially a generated string) to edit it. 

Reviewer 2: ```suggestion
Once the DBMS is shown in the home page, double click on the name (initially a generated string) to edit it. 
```
